test/services/licences/supplementary/process-billing-flag.service.test.js

```diff
@@ -0,0 +1,144 @@
+'use strict'
+
+// Test framework dependencies
+const Lab = require('@hapi/lab')
+const Code = require('@hapi/code')
+const Sinon = require('sinon')
+
+const { describe, it, beforeEach, afterEach } = exports.lab = Lab.script()
+const { expect } = Code
+
+// Test helpers
+const BillRunHelper = require('../../../support/helpers/bill-run.helper.js')
+const ChargeVersionHelper = require('../../../support/helpers/charge-version.helper.js')
+const LicenceHelper = require('../../../support/helpers/licence.helper.js')
+const LicenceSupplementaryYearModel = require('../../../../app/models/licence-supplementary-year.model.js')
+
+// Things we need to stub
+const DetermineChargeVersionYearsService = require('../../../../app/services/licences/supplementary/determine-charge-version-years.service.js')
+
+// Thing under test
+const ProcessSupplementaryBillingFlagService = require('../../../../app/services/licences/supplementary/process-billing-flag.service.js')
+
+describe('Process Billing Flag Service', () => {
+  let chargeVersion
+  let licence
+  let notifierStub
+  let payload
+
+  beforeEach(async () => {
+    // The service depends on GlobalNotifier to have been set. This happens in app/plugins/global-notifier.plugin.js
+    // when the app starts up and the plugin is registered. As we're not creating an instance of Hapi server in this
+    // test we recreate the condition by setting it directly with our own stub
+    notifierStub = { omg: Sinon.stub(), omfg: Sinon.stub() }
+    global.GlobalNotifier = notifierStub
+  })
+
+  afterEach(() => {
+    Sinon.restore()
+  })
+
+  describe('when given a valid payload', () => {
+    describe('with a charge version id', () => {
+      beforeEach(async () => {
+        // region = RegionHelper.select()
+        licence = await LicenceHelper.add()
+        chargeVersion = await ChargeVersionHelper.add({ licenceId: licence.id, endDate: '2023-03-31' })
+
+        payload = {
+          chargeVersionId: chargeVersion.id
+        }
+      })
+
+      describe('that should be flagged for supplementary billing', () => {
+        beforeEach(async () => {
+          Sinon.stub(DetermineChargeVersionYearsService, 'go').resolves({
+            licence: {
+              id: licence.id,
+              regionId: licence.regionId
+            },
+            startDate: chargeVersion.startDate,
+            endDate: chargeVersion.endDate,
+            twoPartTariff: true,
+            flagForBilling: true
+          })
+
+          await BillRunHelper.add({ batchType: 'two_part_tariff', status: 'sent', regionId: licence.regionId })
```

Comment:
Rather than add a bill run to the DB, I would suggest stubbing `DetermineBillRunsSentService` (or whatever it ends up being called!)